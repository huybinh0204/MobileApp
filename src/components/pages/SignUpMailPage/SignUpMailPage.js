import { yupResolver } from '@hookform/resolvers/yup';
import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { useForm } from 'react-hook-form';
import { View } from 'react-native';
import { useSelector } from 'react-redux';
import { object, string } from 'yup';
import { Form, FormInput, FormSubmitButton } from '_components/atoms';
import { OnboardingScreenLayout } from '_components/organisms';
import { NAVIGATION, REGEX } from '_constants';
import strings from '_localization';
import { CustomerSelectors } from '_store/customer';
import { openExternalLink } from '_utilities/ExternalLinks';
import {
  SignUpMailContainer,
  SignUpMailDisclaimerLink,
  SignUpMailDisclaimerText,
  SignUpMailOptionsContainer,
  SignUpMailTitle,
} from './SignUpMailPage.styles';

const schema = object().shape({
  email: string().matches(REGEX.email, strings.signUp.invalidEmail).required(),
});

const defaultValues = {
  email: '',
};

const SignUpMailPage = () => {
  const { navigate } = useNavigation();
  const isLoading = useSelector(CustomerSelectors.getIsLoading);
  const formMethods = useForm({ defaultValues, resolver: yupResolver(schema) });

  const emailValue = formMethods.watch('email');

  const handleDisclaimerLink = async () => {
    await openExternalLink(strings.signUp.disclaimerUrl);
  };

  const handleSubmit = ({ email }) => {
    navigate(NAVIGATION.auth.signUpPassword, { email });
  };

  return (
    <OnboardingScreenLayout step={1} testID="SignUpMailPage">
      <SignUpMailContainer>
        <SignUpMailTitle>{strings.signUp.email.title}</SignUpMailTitle>
        <Form formMethods={formMethods}>
          <FormInput
            allowFontScaling={false}
            autoCapitalize="none"
            autoComplete="email"
            keyboardType="email-address"
            name="email"
            placeholder={strings.signUp.email.placeholder}
            testID="newAccountMail"
            textContentType="emailAddress"
          />
          <SignUpMailOptionsContainer>
            <View>
              <SignUpMailDisclaimerText>
                {strings.signUp.disclaimer}
                <SignUpMailDisclaimerLink onPress={handleDisclaimerLink}>
                  {strings.signUp.disclaimerLink}
                </SignUpMailDisclaimerLink>
              </SignUpMailDisclaimerText>
              <FormSubmitButton
                disabled={emailValue === ''}
                onSubmit={handleSubmit}
                testID="submit"
                isLoading={isLoading}
              >
                {strings.signUp.continue}
              </FormSubmitButton>
            </View>
          </SignUpMailOptionsContainer>
        </Form>
      </SignUpMailContainer>
    </OnboardingScreenLayout>
  );
};

export default SignUpMailPage;
