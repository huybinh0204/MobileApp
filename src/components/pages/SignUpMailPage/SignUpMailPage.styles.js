import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const SignUpMailContainer = styled.View(({ theme }) => ({
  flex: 1,
  justifyContent: 'space-between',
  padding: theme.spacing.m,
}));

export const SignUpMailTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.m,
}));

export const SignUpMailOptionsContainer = styled.View({
  flex: 1,
  justifyContent: 'flex-end',
  width: '100%',
});

export const SignUpMailDisclaimerText = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const SignUpMailDisclaimerLink = styled(BaseText)(({ theme }) => ({
  color: theme.colors.alpha500,
}));
