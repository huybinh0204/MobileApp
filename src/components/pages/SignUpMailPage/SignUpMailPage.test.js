import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { NAVIGATION } from '_constants';
import strings from '_localization/index';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import SignUpMailPage from './SignUpMailPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('SignUpMailPage', () => {
  const mockDispatch = jest.fn();
  const mockNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useDispatch.mockReturnValue(mockDispatch);
    useNavigation.mockReturnValue(mockNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should trigger the change of screen if the mail input is correct', async () => {
    const { getByText, getByTestId } = render(
      <ComponentWithProviders component={SignUpMailPage} />
    );

    const emailInput = getByTestId('newAccountMail');
    fireEvent.changeText(emailInput, 'test@mail.com');

    const sendMail = getByText(strings.signUp.continue);
    fireEvent.press(sendMail);

    await waitFor(() => {
      expect(mockNavigation.navigate).toBeCalledWith(NAVIGATION.auth.signUpPassword, {
        email: 'test@mail.com',
      });
    });
  });

  it('should not navigate if the mail input is not correct', async () => {
    const { getByText, getByTestId } = render(
      <ComponentWithProviders component={SignUpMailPage} />
    );

    const emailInput = getByTestId('newAccountMail');
    fireEvent.changeText(emailInput, 'testmailcom');

    const sendMail = getByText(strings.signUp.continue);
    fireEvent.press(sendMail);

    await waitFor(() => {
      expect(mockNavigation.navigate).toBeCalledTimes(0);
    });
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={SignUpMailPage} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
