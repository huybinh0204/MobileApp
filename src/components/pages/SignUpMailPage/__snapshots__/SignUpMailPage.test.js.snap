// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpMailPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SignUpMailPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "width": "50%",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "backgroundColor": "#F2F3F4",
                                    "borderColor": "#000000",
                                    "borderRadius": 10,
                                    "height": 9,
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "width": "0%",
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "justifyContent": "space-between",
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 36,
                                    "lineHeight": 50.5,
                                    "marginBottom": 58,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              What’s your email?
                            </Text>
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginBottom": 87,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "borderRadius": 6,
                                      "borderWidth": 2,
                                      "flexDirection": "row",
                                      "width": "100%",
                                      "zIndex": 0,
                                    },
                                    Object {
                                      "borderColor": undefined,
                                    },
                                  ]
                                }
                              >
                                <BVLinearGradient
                                  colors={
                                    Array [
                                      309817028,
                                      318208316,
                                    ]
                                  }
                                  endPoint={
                                    Object {
                                      "x": 0.75,
                                      "y": 0.5,
                                    }
                                  }
                                  locations={null}
                                  startPoint={
                                    Object {
                                      "x": 0.25,
                                      "y": 0.5,
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 6,
                                        "height": "100%",
                                        "position": "absolute",
                                        "width": "100%",
                                        "zIndex": 1,
                                      },
                                      Object {
                                        "opacity": undefined,
                                      },
                                    ]
                                  }
                                />
                                <Text
                                  allowFontScaling={false}
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "bottom": 0,
                                        "fontFamily": "Poppins",
                                        "fontSize": 29,
                                        "left": 25.5,
                                        "lineHeight": 40,
                                        "position": "absolute",
                                        "zIndex": 2,
                                      },
                                      Object {
                                        "color": undefined,
                                        "transform": Array [
                                          Object {
                                            "translateY": undefined,
                                          },
                                          Object {
                                            "translateX": undefined,
                                          },
                                          Object {
                                            "scale": undefined,
                                          },
                                        ],
                                      },
                                    ]
                                  }
                                >
                                  Email
                                </Text>
                                <TextInput
                                  allowFontScaling={false}
                                  autoCapitalize="none"
                                  autoComplete="email"
                                  keyboardType="email-address"
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#1F1D25",
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 29,
                                        "lineHeight": 40,
                                        "paddingBottom": 18,
                                        "paddingHorizontal": 25.5,
                                        "paddingTop": 43.5,
                                        "zIndex": 3,
                                      },
                                      undefined,
                                    ]
                                  }
                                  testID="newAccountMail"
                                  textContentType="emailAddress"
                                  value=""
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "justifyContent": "flex-end",
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View>
                                <Text
                                  allowFontScaling={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#5F6A77",
                                        "fontFamily": "Poppins",
                                        "fontSize": 20,
                                        "lineHeight": 28,
                                        "marginBottom": 29,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  By proceeding, you agree to the 
                                  <Text
                                    allowFontScaling={false}
                                    onPress={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#3D3397",
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    Ahead Privacy Policy
                                  </Text>
                                </Text>
                                <View
                                  accessibilityState={
                                    Object {
                                      "disabled": true,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignSelf": "stretch",
                                      "backgroundColor": "#D4D7DB",
                                      "borderRadius": 10,
                                      "opacity": 1,
                                      "paddingHorizontal": 45.5,
                                      "paddingVertical": 31,
                                    }
                                  }
                                  testID="submit"
                                >
                                  <Text
                                    allowFontScaling={false}
                                    disabled={true}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#FFFFFF",
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "textAlign": "center",
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    Continue
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
